forge-gui-mobile/src/forge/adventure/util/RewardActor.java

```diff
@@ -304,6 +310,45 @@ public RewardActor(Reward reward, boolean flippable, RewardScene.Type type) {
                 needsToBeDisposed = true;
                 break;
             }
+            case CardPack: {
+                TextureAtlas atlas = Config.instance().getAtlas(ITEMS_ATLAS);
+                Sprite backSprite = atlas.createSprite("CardBack");
+                if (reward.getDeck() == null) {
+                    needsToBeDisposed = true;
+                    processSprite(backSprite, null, null, 0, 0);
+                    break;
+                }
+
+
+
+                String imageKey;
+                try {
+                    String editionCode = reward.getDeck().getComment();
+                    int artIndex = 1;
+                    if (SealedProduct.specialSets.contains(editionCode) || editionCode.equals("?")) {
+                        imageKey = "b:" + getName().substring(0, getName().indexOf("Booster Pack") - 1);
+                    } else {
+                        int maxIdx = StaticData.instance().getEditions().get(editionCode).getCntBoosterPictures();
+                        artIndex = MyRandom.getRandom().nextInt(maxIdx) + 1;
+                        imageKey = ImageKeys.BOOSTER_PREFIX + editionCode + ((1 >= maxIdx) ? "" : ("_" + artIndex));
+                    }
+                }
+                catch (Exception e){
+                    //Comment did not contain the edition code, this is not a basic booster pack
+                }
+
+
+
+                //FImageComplex cardArt = CardRenderer.getCardArt(imageKey, false, false, false, false, false,false,false,false,false,false);
```

Comment:
this should be generated and drawn via buffers like the tooltip and should be added to dispose method